python/lsst/daf/butler/core/composites.py

```diff
@@ -137,6 +137,8 @@ def shouldBeDisassembled(self, entity: Union[DatasetRef, DatasetType, StorageCla
                 matchName = key
                 break
 
+        assert isinstance(disassemble, bool)
```

Comment:
I should probably just upgrade this to a real exception because it's essentially external config validation.  Of course, it's only the tip of that iceberg, but I figure every little bit helps.